internal/app/mapper/mapper.go

```diff
@@ -192,9 +195,53 @@ func isNotMatch(matchPredicate *matchPredicate, typeURL string, node *core.Node)
 	return !isMatch, nil
 }
 
-func getResult(fragmentRule *rule) string {
-	stringFragment := fragmentRule.GetResult().GetStringFragment()
-	return stringFragment
+func getResult(fragmentRule *rule, node *core.Node) (string, error) {
+	found, result, err := getResultFromRequestNodeFragment(fragmentRule, node)
+	if err != nil {
+		return "", err
+	}
+	if found {
+		return result, nil
+	}
+
+	return fragmentRule.GetResult().GetStringFragment(), nil
+}
+
+func getResultFromRequestNodeFragment(fragmentRule *rule, node *core.Node) (bool, string, error) {
+	requestNodeFragment := fragmentRule.GetResult().GetRequestNodeFragment()
+	if requestNodeFragment == nil {
+		return false, "", nil
+	}
+
+	nodeField := requestNodeFragment.GetField()
+	var nodeValue = ""
+	switch nodeField {
+	case aggregationv1.NodeFieldType_NODE_CLUSTER:
+		nodeValue = node.GetCluster()
+	case aggregationv1.NodeFieldType_NODE_ID:
+		nodeValue = node.GetId()
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_REGION:
+		nodeValue = node.GetLocality().GetRegion()
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_ZONE:
+		nodeValue = node.GetLocality().GetZone()
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_SUBZONE:
+		nodeValue = node.GetLocality().GetSubZone()
+	}
```

Comment:
Can you add a default and error?

Reply:
done.

Reply:
shouldn't we assume that the proto will be valid at this point?

Reply:
That is true, but imo its good to have functional completeness of the component.

Reply:
Also for if we ever change the proto and add new type support, but forget to update the mapper